lib/kitchen/directions/bake_index/v1.rb

```diff
@@ -118,10 +118,25 @@ def bake(book:)
       @metadata_elements = book.metadata.children_to_keep.copy
       @index = Index.new
 
-      book.pages.terms.each do |term_element|
+      chapters = book.chapters
+      book.pages_and_composite_pages.terms.each do |term_element|
         # Markup the term
-        page = term_element.ancestor(:page)
-        term_element.id = "auto_#{page.id}_term#{term_element.count_in(:book)}"
+        begin
+          page = term_element.ancestor(:page)
+        rescue RuntimeError
+          page = term_element.ancestor(:composite_page)
+        end
+        term_element.id = ''
+        page_title = ''
+        if page.instance_of?(Kitchen::CompositePageElement)
+          term_element.id = "auto_composite_page_term#{term_element.count_in(:book)}"
+          chapter_number = chapters.search("[id='#{term_element.id}']")&.first&.ancestor(:chapter)&.count_in(:book)
+          title_text = page.title.text.strip
+          page_title = "#{chapter_number} #{title_text}".strip
+        else
+          term_element.id = "auto_#{page.id}_term#{term_element.count_in(:book)}"
+          page_title = page.title.text
+        end
```

Comment:
Just realized this other way to make this cleaner -- two loops -- would mean the term counts in book aren't unique, but the overall ID would still be

```ruby
def bake(book:)
  ...
  book.pages.terms.each do |term_element|
    page = term_element.ancestor(:page)
    term_element.id = "auto_#{page.id}_term#{term_element.count_in(:book)}"
    page_title = page.title.text
    add_to_index(term_element, page_title)
  end
  
  book.chapters.composite_pages.terms.each do |term_element|
    page = term_element.ancestor(:composite_page)
    chapter = term_element.ancestor(:chapter)
    term_element.id = "auto_composite_page_term#{term_element.count_in(:book)}"
    title_text = page.title.text.strip
    page_title = "#{chapter.count(in: :book)} #{title_text}".strip
    add_to_index(term_element, page_title)
  end
  ...
end

def add_to_index(term_element, page_title)
  group_by = term_element.text.strip[0]
  group_by = I18n.t(:eob_index_symbols_group) unless group_by.match?(/\w/)
  term_element['group-by'] = group_by

  # Add it to our index object
  @index.add_term(
    Term.new(
      text: term_element.text,
      id: term_element.id,
      group_by: group_by,
      page_title: page_title.gsub(/\n/, '')
    )
  )
end
```